internal/linter/check_reserved.go

```diff
@@ -0,0 +1,98 @@
+package linter
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/skeema/skeema/internal/tengo"
+)
+
+func init() {
+	RegisterRule(Rule{
+		CheckerFunc:     TableBinaryChecker(reservedChecker),
```

Comment:
Regular `TableChecker` is better here, since that can return multiple annotations; in theory a single table might have multiple reserved words.

The various Checker types are basically just syntactic sugar to make it easier to write linter check functions that operate on specific types and return one or multiple annotations -- see https://github.com/skeema/skeema/blob/94eaa99/internal/linter/linter.go#L41:L99